modal/sandbox.py

```diff
@@ -655,7 +656,16 @@ async def exec(
         )
         resp = await retry_transient_errors(self._client.stub.ContainerExec, req)
         by_line = bufsize == 1
-        return _ContainerProcess(resp.exec_id, self._client, stdout=stdout, stderr=stderr, text=text, by_line=by_line)
+        exec_deadline = time.monotonic() + timeout if timeout else None
```

Comment:
As discussed we need to add a tolerance here for latency between the client and the worker to avoid missing messages sent before the specificed timeout on the worker side but which are received after the specified timeout on the client side.